### PR TITLE
Allow the user to select containers when adding a secret to an application

### LINKED ISSUE #2024: Allow the user to select containers when adding a secret to an application

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
Fixes #2024

image

image

### CODE REVIEW

File: app/views/directives/add-secret-to-application.html [hunk 1]
```diff
@@ -69,6 +69,23 @@
               </div>
             </div>
           </div>
+          <div ng-if="ctrl.addType === 'volume' && ctrl.application.spec.template.spec.containers.length > 1"  class="volume-options">
```
Comment (spadgett) [inline app/views/directives/add-secret-to-application.html]: We should go ahead do the same for environment variables as well I think

File: app/views/directives/add-secret-to-application.html [hunk 2]
```diff
@@ -69,6 +69,23 @@
...
+            <h4>Containers</h4>
+            <div ng-if="ctrl.attachAllContainers">
+              The volume will be mounted into all containers. You can
+              <a href="" ng-click="ctrl.attachAllContainers = false">select specific containers</a>
+              instead.
+            </div>
+            <div ng-if="!ctrl.attachAllContainers" class="form-group">
```
Comment (spadgett) [inline app/views/directives/add-secret-to-application.html]: @jeff-phillips-18 Is there a max-height on the dialog in case there are a lot of containers?

Comment (jeff-phillips-18) [reply app/views/directives/add-secret-to-application.html]: nope, I suppose there ought to be. Thanks!

File: app/views/directives/add-secret-to-application.html (hunk 1 above)
Comment (jeff-phillips-18) [reply app/views/directives/add-secret-to-application.html]: Ah, OK

Comment (jeff-phillips-18) [reply app/views/directives/add-secret-to-application.html]: Done

File: app/views/directives/add-secret-to-application.html (hunk 2 above)
Comment (jeff-phillips-18) [reply app/views/directives/add-secret-to-application.html]: Set the max height on the list of containers.

**Comment by jeff-phillips-18** [comment]
Updates to allow container selection for both Environment Variables and Volume

image

with scrolling containers:

image

**Comment by spadgett** [comment]
[merge][severity: bug]

**Comment by spadgett** [comment]
[merge][severity: bug]

**Pull request merged**